modules/redshift-loader/src/main/scala/com/snowplowanalytics/snowplow/loader/redshift/Redshift.scala

```diff
@@ -58,26 +50,45 @@ object Redshift {
-            val ddlFile = MigrationGenerator.generateMigration(migration, 4096, Some(schema))
-
-            val (preTransaction, inTransaction) = ddlFile.statements.foldLeft((NoPreStatements, NoStatements)) {
-              case ((preTransaction, inTransaction), statement) =>
-                statement match {
-                  case s @ AlterTable(_, _: AlterType) =>
-                    (Item.AlterColumn(Fragment.const0(s.toDdl)) :: preTransaction, inTransaction)
-                  case s @ AlterTable(_, _) =>
-                    (preTransaction, Item.AddColumn(Fragment.const0(s.toDdl), ddlFile.warnings) :: inTransaction)
-                  case _ => // We explicitly support only ALTER TABLE here; also drops BEGIN/END
-                    (preTransaction, inTransaction)
-                }
-            }
-
-            val target = SchemaKey(migration.vendor, migration.name, "jsonschema", migration.to)
-            Block(preTransaction.reverse, inTransaction.reverse, Entity.Table(schema, target))
+          override def updateTable(shredModel: ShredModel.GoodModel, currentSchemaKey: SchemaKey): Block = {
+            val outTransactions = shredModel.migrations.outTransaction(Some(currentSchemaKey))
+            val inTransactions = shredModel.migrations.inTransaction(Some(currentSchemaKey))
+            val outTransactionToSql =
+              outTransactions.map { varcharExtension =>
+                s"""  ALTER TABLE $schema.${shredModel.tableName}
+                 |    ALTER COLUMN "${varcharExtension.old.columnName}" TYPE ${varcharExtension.newEntry.columnType.show};
+                 |""".stripMargin
+              }.mkString
+            val inTransactionToSql =
+              inTransactions.map { columnAddition =>
+                s"""  ALTER TABLE $schema.${shredModel.tableName}
+                   |    ADD COLUMN "${columnAddition.column.columnName}" ${columnAddition.column.columnType.show} ${columnAddition.column.compressionEncoding.show};
+                   |""".stripMargin
+              } match {
+                case Nil => s"""|
+                                |-- NO ADDED COLUMNS CAN BE EXPRESSED IN SQL MIGRATION
+                                |
+                                |COMMENT ON TABLE $schema.${shredModel.tableName} IS '${shredModel.schemaKey.toSchemaUri}';
+                                |""".stripMargin
+                case h :: t => s"""|
+                                  |BEGIN TRANSACTION;
+                                   |
+                                   |${(h :: t).mkString}
+                                   |  COMMENT ON TABLE $schema.${shredModel.tableName} IS '${shredModel.schemaKey.toSchemaUri}';
+                                   |
+                                   |END TRANSACTION;""".stripMargin
+              }
+            val preTransaction =
+              if (outTransactions.nonEmpty) Item.AlterColumn(Fragment.const0(outTransactionToSql)) :: Nil
```

Comment:
On this line you create a list of a single `Item.AlterColumn`.  But when you look closely, that single item is comprised of multiple sql statements which you have joined together using `;`.

I think this is the wrong way to create a `AlterColumn`.  The correct way is to return a list of multiple `Item.AlterColumn` where each item represents a single column.

There is a good reason for changing this:  We have a feature in which RDB loader recovers from previous failed attempts to alter the table.  This feature handles the case where some pre-migrations were successful and some pre-migrations had a failure, which has left the table in an inconsistent state.  But this recovery feature will not work if the alter table statements are concatenated together into a single string.